heisskleber/mqtt/publisher_async.py

```diff
@@ -45,7 +44,7 @@ async def send_work(self) -> None:
         while True:
```

Comment:
While I understand the pattern, it seems a little "wrong" to have two while True loops to capture a dying connection - maybe with the next refactor move to a callback pattern? or raise an Exception and let the whole thing die gracefully to make sure, the dropped connection is actually noted outside of the execution loop? Consider the following scenario: due to misconfiguration, the connection times out upon instantiation, but due to the pattern, instead of raising an Exception, the connection will be opened again and again, while any service relying on the code will never know, that the connection will never actually work.

Reply:
I think the proper approach would be to log the exception. But I'm against raising the Exception, as a failed connection would bring down an entire program, when you can just reconnect. e.g. the broker restarts and all connecting clients error out, that's not desirable.

But possibly a "unrecoverable disconnect" could be defined which would trigger an Exception.
I like the callback idea as well, that gives anyone who uses the library freedom to handle disconnects however they like (i.e. don't implement it and it will keep trying forever or shut down the program after n disconnects).